tdmq/client/timeseries.py

```diff
@@ -0,0 +1,70 @@
+import abc
+import numpy as np
+
+
+class TimeSeries(abc.ABC):
+
+    def pre_fetch(self):
+        args = {'after': self.after, 'before': self.before,
+                'bucket': self.bucket, 'op': self.op}
+        res = self.sensor.get_timeseries(args)
+        self.timebase = res['timebase']
+        self.timedelta = np.array(res['timedelta'])
+        return res['data']
+
+    @abc.abstractmethod
+    def fetch(self):
+        pass
+
+    @abc.abstractmethod
+    def get_item(self, args):
+        pass
+
+    def __getitem__(self, indx):
+        return self.get_item(np.index_exp[indx])
+
+    def __init__(self, sensor, after, before, bucket, op):
+        self.sensor = sensor
+        self.after = after
+        self.before = before
+        self.bucket = bucket
+        self.op = op
+        self.fetch()
```

Comment:
I would make the `fetch` separate from object construction.  This would allow the existence of instances that represent a query that hasn't been executed.  It might also be useful for unit testing.